src/Skia/Avalonia.Skia/DrawingContextImpl.cs

```diff
@@ -189,7 +189,10 @@ public void DrawBitmap(IBitmapImpl source, double opacity, Rect sourceRect, Rect
             var d = destRect.ToSKRect();
 
             var paint = SKPaintCache.Shared.Get();
-            paint.Color = new SKColor(255, 255, 255, (byte)(255 * opacity * (_useOpacitySaveLayer ? 1 : _currentOpacity)));
+
+            var useOpacitySaveLayer = _useOpacitySaveLayer || RenderOptions.RequiresFullOpacityHandling == true;
```

Comment:
We need to push the `_currentOpacity` value and set it to 1. Otherwise rendering will be broken with the following visual tree structure:
```
- Border №1 - opacity: 0.8
  - Border №2 - opacity: 0.8, requires full opacity: true
     - Border №3 - opacity: 0.8, requires full opacity: true
```  
In this case Border №3 would get rendered with 0.64 opacity into a layer created by Border №2, while it should be rendered with 0.8.

Reply:
Another problem I think we have is the first two borders: we don't take the accumulated _currentOpacity from Border №1 when rendering the layer from Border №2.